File: insertion-sort.js
const arr = [3, 4, 30, 1, 45, 34, 1094, 6, 4, 90, 84, 2];


for (let j = 1; j < arr.length; j++) {
    const key = arr[j];
    let i = j - 1;
    while (i >= 0 && arr[i] > key) {
        arr[i+1] = arr[i];
        i -= 1;
    }
    arr[i + 1] = key;
}
console.log(arr);

File: rust-algorithms/src/algorithms/sort/insertion_sort.rs
pub fn insertion_sort(arr: &mut [i32]) {
    for i in 1..arr.len() {
        let mut j = i;
        while j > 0 && arr[j - 1] > arr[j] {
            arr.swap(j - 1, j);
            j -= 1;
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_insertion_sort() {
        let mut arr = [1, 5, 2, 4, 3];
        insertion_sort(&mut arr);
        assert_eq!(arr, [1, 2, 3, 4, 5]);
    }
}
